samcli/lib/config/file_manager.py

```diff
@@ -0,0 +1,187 @@
+        """
+        raise NotImplementedError("Write method not implemented.")
+
+    @staticmethod
+    @abstractmethod
+    def write_document(document: Any, filepath: Path):
+        """
+        Write the contents of a document object to a file at the provided location.
+
+        Parameters
+        ----------
+        document: Any
+            The object to write.
+        filepath: Path
+            The final location for the file to be written.
+        """
+        raise NotImplementedError("Write document method not implemented.")
+
+    @staticmethod
+    @abstractmethod
+    def put_comment(document: Any, comment: str) -> Any:
+        """
+        Put a comment in a document object.
+
+        Parameters
+        ----------
+        document: Any
+            The object to write
+        comment: str
+            The comment to include in the document.
+
+        Returns
+        -------
+        Any
+            The new document, with the comment added to it.
+        """
+        raise NotImplementedError("Put comment method not implemented.")
+
+
+class TomlFileManager(FileManager):
+    """
+    Static class to read and write toml files.
+    """
+
+    @staticmethod
+    def read(filepath: Path) -> Any:
+        """
+        Read a TOML file at the given path.
+
+        Parameters
+        ----------
+        filepath: Path
+            The Path object that points to the file to be read.
+
+        Returns
+        -------
+        Any
+            A dictionary-like tomlkit.TOMLDocument object, which represents the contents of the TOML file at the
+            provided location.
+        """
+        toml_doc = tomlkit.document()
+        try:
+            txt = filepath.read_text()
+            toml_doc = tomlkit.loads(txt)
+        except OSError as e:
+            LOG.debug(f"OSError occurred while reading TOML file: {str(e)}")
+        except tomlkit.exceptions.TOMLKitError as e:
+            raise FileParseException(e) from e
+
+        return toml_doc
+
+    @staticmethod
+    def write(document: dict, filepath: Path):
+        """
+        Write the contents of a dictionary to a TOML file at the provided location.
+
+        Parameters
+        ----------
+        document: dict
+            The object to write.
+        filepath: Path
+            The final location for the TOML file to be written.
+        """
+        if not document:
+            LOG.debug("No document given for TomlFileManager to write.")
+            return
+
+        doc_no_comments = {k: v for k, v in document.items() if k != "__comment__"}
+        toml_doc = tomlkit.document()
+
+        if document.get("__comment__", None):  # Comment appears at the top of doc
+            toml_doc.add(tomlkit.comment(document["__comment__"]))
+        for k, v in doc_no_comments.items():
+            toml_doc.add(k, v)
+
+        filepath.write_text(tomlkit.dumps(toml_doc))
+
+    @staticmethod
+    def write_document(document: Any, filepath: Path):
```

Comment:
Can we use specific typing instead of `Any`, `document: TOMLDocument`? Makes it easier to for type checking to catch any errors.

Reply:
Yup, in my changes just pushed, we set it to be generic `dict` since the `TOMLDocument` is a subclass of dictionaries.

Reply:
We could change to be the specific `TOMLDocument`, but then the method signature differs from the abstract class it inherits from